build/kj_test.bzl

```diff
@@ -6,18 +6,55 @@ def kj_test(
         size = "medium",
         **kwargs):
     test_name = src.removesuffix(".c++")
-    native.cc_test(
-        name = test_name,
-        srcs = [src],
-        deps = [
-            "@capnp-cpp//src/kj:kj-test",
-        ] + deps,
-        linkopts = select({
-            "@//:use_dead_strip": ["-Wl,-dead_strip", "-Wl,-no_exported_symbols"],
-            "//conditions:default": [""],
-        }),
-        data = data,
-        tags = tags,
-        size = size,
-        **kwargs
+
+    # Create wrapper source file that will inject autogate initialization based on environment variable
+    wrapper_name = test_name + "_autogate_wrapper"
+    wrapper_file = wrapper_name + ".c++"
+
+    native.genrule(
```

Comment:
Wouldn't it be simpler to put some code in `autogate.c++` which checks for the env var when it finds the autogates uninitialized in `isEnabled()`?

Reply:
I just saw this. You're right.